File: app/src/main/java/eu/kanade/mangafeed/data/database/models/MangaCategory.java
package eu.kanade.mangafeed.data.database.models;

import com.pushtorefresh.storio.sqlite.annotations.StorIOSQLiteColumn;
import com.pushtorefresh.storio.sqlite.annotations.StorIOSQLiteType;

import eu.kanade.mangafeed.data.database.tables.MangaCategoryTable;

@StorIOSQLiteType(table = MangaCategoryTable.TABLE)
public class MangaCategory {

    @StorIOSQLiteColumn(name = MangaCategoryTable.COLUMN_ID, key = true)
    public Long id;

    @StorIOSQLiteColumn(name = MangaCategoryTable.COLUMN_MANGA_ID)
    public long manga_id;

    @StorIOSQLiteColumn(name = MangaCategoryTable.COLUMN_CATEGORY_ID)
    public int category_id;

    public MangaCategory() {}

    public static MangaCategory create(Manga manga, Category category) {
        MangaCategory mc = new MangaCategory();
        mc.manga_id = manga.id;
        mc.category_id = category.id;
        return mc;
    }

}

File: app/src/main/java/eu/kanade/tachiyomi/data/database/models/MangaCategory.kt
package eu.kanade.tachiyomi.data.database.models

import eu.kanade.domain.category.model.Category as DomainCategory
import eu.kanade.domain.manga.model.Manga as DomainManga

class MangaCategory {

    var id: Long? = null

    var manga_id: Long = 0

    var category_id: Int = 0

    companion object {

        fun create(manga: DomainManga, category: DomainCategory): MangaCategory {
            val mc = MangaCategory()
            mc.manga_id = manga.id
            mc.category_id = category.id.toInt()
            return mc
        }

        fun create(manga: Manga, category: Category): MangaCategory {
            val mc = MangaCategory()
            mc.manga_id = manga.id!!
            mc.category_id = category.id!!
            return mc
        }
    }
}
